File: IndexOfAnExtraElement/IndexOfAnExtraElement.cpp
/*
https://www.geeksforgeeks.org/problems/index-of-an-extra-element/1
*/

//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    
    int binarySearch(int n, int arr1[], int arr2[]){
        int left=0, right=n-2, res=n-1;
        while(left<=right){
            int mid=left+(right-left)/2;
            if(arr2[mid]>arr1[mid]){
                res=mid;
                right=mid-1;
            } else if(arr2[mid]<arr1[mid]){
                right=mid-1;
            } else left=mid+1;
        }
        return res;
    }
    
    int findExtra(int n, int arr1[], int arr2[]) {
        return binarySearch(n, arr1, arr2);
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int arr1[n], arr2[n - 1];
        for (int i = 0; i < n; i++) {
            cin >> arr1[i];
        }
        for (int i = 0; i < n - 1; i++) {
            cin >> arr2[i];
        }
        Solution obj;
        cout << obj.findExtra(n, arr1, arr2) << endl;
    }
}
// } Driver Code Ends

File: IndexOfAnExtraElement/IndexOfAnExtraElement.java
/*
https://www.geeksforgeeks.org/problems/index-of-an-extra-element/1
*/

//{ Driver Code Starts
import java.util.*;

class ExtraElement {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-- > 0) {
            int n = sc.nextInt();
            int[] a = new int[n];
            int[] b = new int[n - 1];
            for (int i = 0; i < n; i++) a[i] = sc.nextInt();
            for (int i = 0; i < n - 1; i++) b[i] = sc.nextInt();
            Solution g = new Solution();
            System.out.println(g.findExtra(n, a, b));
        }
    }
}
// } Driver Code Ends


/*Complete the function below*/
class Solution {
    
    public static int binarySearchForIndexOfExtraElementInArr1(int n, int arr1[], int arr2[]){
        int currentLeftIndexInArr2=0, currentRightIndexInArr2=n-2, resultantIndexOfExtraElementInArr1=n-1;
        while(currentLeftIndexInArr2<=currentRightIndexInArr2){
            int currentMiddleIndexInArr2=currentLeftIndexInArr2+(currentRightIndexInArr2-currentLeftIndexInArr2)/2;
            if(arr2[currentMiddleIndexInArr2]>arr1[currentMiddleIndexInArr2]){
                resultantIndexOfExtraElementInArr1=currentMiddleIndexInArr2;
                currentRightIndexInArr2=currentMiddleIndexInArr2-1;
            } else if(arr2[currentMiddleIndexInArr2]<arr1[currentMiddleIndexInArr2]){
                currentRightIndexInArr2=currentMiddleIndexInArr2-1;
            } else currentLeftIndexInArr2=currentMiddleIndexInArr2+1;
        }
        return resultantIndexOfExtraElementInArr1;
    }
    
    public int findExtra(int n, int arr1[], int arr2[]) {
        int resultantIndexOfExtraElementInArr1=binarySearchForIndexOfExtraElementInArr1(n, arr1, arr2);
        return resultantIndexOfExtraElementInArr1;
    }
}